### PR TITLE
Add toYaml & fromYaml functions

### PR DESCRIPTION
**Pull request by jan-di**
### Description of your changes

Add two functions `toYaml` and `fromYaml`to this function, as they are not supported in sprig. Esepcially the first one is essential, as the templates are focused on creating yaml resources.

I have:

- [x] Read and followed Crossplane's [contribution process].
- ~[ ] Added or updated unit tests for my change.~

[contribution process]: https://git.io/fj2m9
[docs]: https://docs.crossplane.io/contribute/contribute

### CODE REVIEW

**Comment by tnm-** [comment]
Thx, I was looking for exactly this. Without it, i believe go templates need to be aware of the input schema all the way down even to copy values from those fields (i.e. to not change them at all).

**Comment by ezgidemirel** [comment]
Hi @jan-di , thanks for extending available helper functions! Can you please create a new folder under `examples` and provide manifests that can work with `crossplane beta render` tooling?

**Comment by jan-di** [comment]
@ezgidemirel done. Please note I also added `example/.dev/functions.yaml` to be able to run pipelines locally using go instead of using a tagged image.

**Comment by tnm-** [comment]
Could merging and releasing this get some priority? I am aware that functions are still beta feature, and there's no bug report for this (yes, I consider the lack of this a bug, not an enhancement -- should we open an bug type issue as well?). Without this, function-go-templating is very limited in usability, unfortunately.

Thank you!

File: README.md [hunk 1]
```diff
@@ -98,6 +98,14 @@ $ crossplane beta render xr.yaml composition.yaml functions.yaml
 See the [composition functions documentation][docs-functions] to learn more
 about `crossplane beta render`.
 
+## Additional functions
+
+| Name                                              | Description                             |
+| ------------------------------------------------- | --------------------------------------- |
+| [`randomChoice``](example/functions/randomChoice) | Randomly selects one of a given strings |
```
Comment (ezgidemirel) [inline README.md:105]:
```suggestion
| [`randomChoice`](example/functions/randomChoice) | Randomly selects one of a given strings |
```

**Comment by phisco** [comment]
`toYaml` shouldn't be strictly required, as JSON is totally valid inside YAML, surely not the best to see in the output of the render subcommands, but for what crossplane is concerned, it should just work.

`fromYaml` could indeed give you access to YAML passed as string as per your example, so although nice to have, I wouldn't call it a bug, @tnm-.

**Review by ezgidemirel — APPROVED** [review]
Thanks @jan-di for the PR! Added a small comment to fix README. Apart from that, there are some conflicts that need to be resolved. Could you please rebase your PR, so that we can merge it :)

**Comment by tnm-** [comment]
> `toYaml` shouldn't be strictly required, as JSON is totally valid inside YAML, surely not the best to see in the output of the render subcommands, but for what crossplane is concerned, it should just work.

I disagree, I think toYaml is required. Or at least I haven't figured out how to take arbitrary yaml object from claim/xr and output it to the managed resource verbatim (either as yaml or json) without knowing and defining the exact schema of the arbitrary yaml object in the go template.

**Comment by jan-di** [comment]
> Thanks @jan-di for the PR! Added a small comment to fix README. Apart from that, there are some conflicts that need to be resolved. Could you please rebase your PR, so that we can merge it :)

Thank you for the review. From my side, everything is done and green :)

**Pull request merged**